cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsStorage.java

```diff
@@ -36,4 +37,22 @@ public StorageType.Type getType() {
   public StorageClient<?> getClient() {
     return hdfsStorageClient;
   }
+
+  /**
+   * Allocates Table Space for the HDFS storage.
+   *
+   * <p>tableLocation looks like: /{rootPrefix}/{databaseId}/{tableId}-{tableUUID} We strip the
+   * endpoint as it's not needed for HDFS.
+   *
+   * @return the table location
+   */
+  @Override
+  public String allocateTableSpace(
+      String databaseId,
+      String tableId,
+      String tableUUID,
+      String tableCreator,
+      boolean skipProvisioning) {
+    return Paths.get(getClient().getRootPrefix(), databaseId, tableId + "-" + tableUUID).toString();
```

Comment:
The BasedStorage implementation did not use `java.nio.file.Paths`. So why it has to be done differently?

Reply:
Good observation, we do this to ensure backward-compatibility with current implementation. the difference with baseStorage class is the no-endpoint, basically baseStorage gives `hdfs://data/openhouse` whereas the override here gives `/data/openhouse`